src/data_types/nonce.rs

```diff
@@ -187,9 +196,22 @@ impl<'a> Deserialize<'a> for Nonce {
             {
                 Nonce::from_dec(value).map_err(E::custom)
             }
+
+            fn visit_seq<E>(self, mut seq: E) -> Result<Self::Value, E::Error>
+            where
+                E: SeqAccess<'a>,
+            {
+                let mut vec = Vec::new();
+
+                while let Ok(Some(Value::Number(elem))) = seq.next_element() {
+                    vec.push(elem.as_u64().unwrap() as u8);
```

Comment:
This should likely be more tolerant of bad inputs, not panicking via `unwrap`. It should also return an error if a non-Number element is found.

Reply:
fixed